src/lib.rs

```diff
@@ -982,7 +1000,8 @@ impl ProgramVersion<'_> {
         &mut self,
         mut program_blueprint: ProgramBlueprint,
         channel: Channel,
-        event_buffer_size: usize,
+        buffer_capacity: usize,
+        element_size: usize,
```

Comment:
I'm not a huge fan of passing stuff down like this but we're already doing it and I can't think of a better way.